src/reporter/annoations-reporter.ts

```diff
@@ -0,0 +1,96 @@
+/*
+   Copyright 2022 Mitch Spano
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+	 https://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+import { getScannerViolationType } from "../common";
+import { Octokit } from "@octokit/action";
+import { context } from "@actions/github";
+import {
+  BaseReporter,
+  GithubAnnotation,
+  GithubCheckRun,
+} from "./reporter.types";
+import { ScannerViolation } from "../sfdxCli";
+
+export const ERROR = "Error";
+export const RIGHT = "RIGHT";
+
+export class AnnotationsReporter extends BaseReporter<GithubAnnotation> {
```

Comment:
Can we add jsdoc formatted comments on this class and all of its methods?

Reply:
Done